ext/afform/core/ang/af/afFieldset.directive.js

```diff
@@ -86,16 +86,45 @@
 
         /**
          * Get fieldset values to use for afform filters
+         *
+         * Filter out empty values (null, '', [], {}) at top level and one level
+         * below for object values (fields with ranges/operators)
+         *
          * @returns Object
          */
         this.getFilterValues = () => {
-          const data = this.getFieldData();
-          // filter out unset values
-          // intended to be equivalent to previous lodash implementation
-          // (typeof val !== 'undefined' && val !== null && (_.includes(['boolean', 'number', 'object'], typeof val) || val.length));
-          return Object.fromEntries(Object.entries(data).filter(([key, value]) =>
-            (['boolean', 'number', 'object'].includes(typeof value) && value !== null) || (value && value.length)
-          ));
+          // find null, '', [], {}
+          const isEmpty = (v) => {
+            if (v === null || v === '') {
+              return true;
+            }
+            if (v && v.length === 0) {
```

Comment:
What if `v` is a primitive without a `length` property?

Should this be
```suggestion
            if (Array.isArray(v) && v.length === 0) {
```